libcontainer/configs/validate/validator.go

```diff
@@ -42,6 +42,7 @@ func Validate(config *configs.Config) error {
 	// Relaxed validation rules for backward compatibility
 	warns := []check{
 		mountsWarn,
+		rootlessSharedPidns, // TODO: make it an error in runc 1.3.
```

Comment:
No need to error out if we implement walking the process tree

Reply:
> No need to error out if we implement walking the process tree

To me it looks neither possible nor desirable.

Not possible because
 - init process might be gone already (with some other processes still running);
 - walking the tree requires freezing all the processes, as otherwise the walker will be racing with forks;

Not desirable because
 - the code will probably be very slow and resource hungry, thanks to text-based nature of /proc;
 - cgroup v1 is going to be obsolete (one day it will; fingers crossed).

Reply:
Also, the only possible way to implement this process tree walking nicely (i.e. not slow and resource hungry) would be via ebpf (which has access to kernel-internal data structures), but the kernels supporting it are probably running cgroup v2 already. Even with ebpf, other issues cited above remain.